tests/SpecTests/CrudSpecTest.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace MongoDB\Tests\SpecTests;
+
+use stdClass;
+
+/**
+ * Transactions spec tests.
+ *
+ * @see https://github.com/mongodb/specifications/tree/master/source/crid
+ */
+class CrudSpecTest extends FunctionalTestCase
+{
+    /* These should all pass before the driver can be considered compatible with
+     * MongoDB 4.2. */
+    private static $incompleteTests = [
+        'aggregate-merge: Aggregate with $merge' => 'PHPLIB-438',
+        'aggregate-merge: Aggregate with $merge and batch size of 0' => 'PHPLIB-438',
+        'aggregate-merge: Aggregate with $merge and majority readConcern' => 'PHPLIB-438',
+        'aggregate-merge: Aggregate with $merge and local readConcern' => 'PHPLIB-438',
+        'aggregate-merge: Aggregate with $merge and available readConcern' => 'PHPLIB-438',
+        'aggregate-out-readConcern: readConcern majority with out stage' => 'PHPLIB-431',
+        'aggregate-out-readConcern: readConcern local with out stage' => 'PHPLIB-431',
+        'aggregate-out-readConcern: readConcern available with out stage' => 'PHPLIB-431',
+        'aggregate-out-readConcern: readConcern linearizable with out stage' => 'PHPLIB-431',
+        'aggregate-out-readConcern: invalid readConcern with out stage' => 'PHPLIB-431',
+        'bulkWrite-arrayFilters: BulkWrite with arrayFilters' => 'Fails due to command assertions',
+        'updateWithPipelines: UpdateOne using pipelines' => 'PHPLIB-418',
+        'updateWithPipelines: UpdateMany using pipelines' => 'PHPLIB-418',
+        'updateWithPipelines: FindOneAndUpdate using pipelines' => 'PHPLIB-418',
+    ];
+
+    /**
+     * Assert that the expected and actual command documents match.
+     *
+     * Note: this method may modify the $expected object.
+     *
+     * @param stdClass $expected Expected command document
+     * @param stdClass $actual   Actual command document
+     */
+    public static function assertCommandMatches(stdClass $expected, stdClass $actual)
+    {
+        /* The command may contain a session id, even though we don't start one.
+         * In this case, ignore any session identifier by removing it. */
+        if (isset($actual->lsid)) {
+            unset($actual->lsid);
+        }
```

Comment:
Can you confirm if this is necessary? `assertDocumentsMatch()` should ignore extra fields in the actual document (see doc block in FunctionalTestCase as well as DocumentsMatchConstraint for more info).

I'll note that the change stream spec, which doesn't assert `lsid` at all, didn't need this.

Reply:
This might have been left over from me trying to get the bulkWrite test to pass. `assertDocumentsMatch` decided that those documents didn't match, but I'll have to re-check that anyways. I'll remove this piece of specific code and fix the bulkWrite spec test in a separate PR.